docs/src/main/asciidoc/_high-availability/cloud/Kubernetes.adoc

```diff
@@ -0,0 +1,236 @@
+// The provided expiration configuration saves unnecessary 1 expiration thread that would otherwise run.
+
+[source,cli,title=infinispan.cli]
+----
+/subsystem=infinispan/cache-container=web/distributed-cache=sessions:add
+/subsystem=infinispan/cache-container=web/distributed-cache=sessions/component=expiration:add(interval=0)
+/subsystem=infinispan/cache-container=web:write-attribute(name=default-cache,value=sessions)
+----
+
+The Maven plugin configuration can then be updated to update the generated container image:
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-maven-plugin</artifactId>
+    <executions>
+        <execution>
+            <!-- ... -->
+            <configuration>
+                <!-- ... -->
+                <packaging-scripts>
+                    <packaging-script>
+                        <scripts>
+                            <script>infinispan.cli</script>
+                        </scripts>
+                    </packaging-script>
+                </packaging-scripts>
+            </configuration>
+        </execution>
+    </executions>
+</plugin>
+----
+
+== Ingress Configuration
+
+A https://kubernetes.io/docs/concepts/services-networking/ingress/[Kubernetes Ingress] is an object that manages external access to the services in a cluster, typically via HTTP.
+It provides load balancing, SSL termination and name-based virtual hosting.
+In order to support the affinity requirements described above, an Ingress Controller implementation, supporting L7 load balancing and exposes the requisite annotations for configuring session affinity, will need to be configured.
+The following sections will configure an https://haproxy-ingress.github.io/[HAProxy Ingress Controller] for optimal routing.
+
+=== Configuring Affinity in Deployment Image
+
+// TODO - update this section as we make this the default - https://issues.redhat.com/browse/CLOUD-4174
+
+First, we will need to reconfigure the application server to store the affinity information in a separate cookie.
+Previously WildFly only supported the ability to encode session affinity information into the JSESSIONID cookie, which is already used to uniquely identify a session.
+While this works well with the httpd family of load balancer modules (e.g. mod_cluster, mod_proxy_balancer, mod_jk) for which this mechanism was designed,
+the current set of load balancers used in the cloud rely on a separate cookie for implementing sticky sessions (e.g. HAProxy),
+and thus the default WildFly configuration lacks the ability to allow WildFly to guide requests for a given session to the server that can most efficiently handle it.
+To resolve this, we need to configure the Undertow subsystem to use a separate cookie to send affinity information.
+Now that the configured cache is using the distribution mode, the load balancer will need to be made aware of the affinity.
```

Comment:
Just to say, guiding requests to the primary owner is relevant to replicated caches as well.

Reply:
Good point - reworded.